ArchiSteamFarm/Bot.cs

```diff
@@ -2836,6 +2839,295 @@ public sealed class Bot : IAsyncDisposable {
+
+					for (byte page = 2; page <= maxPages; page++) {
+						// We need a copy of variable being passed when in for loops, as loop will proceed before our task is launched
+						byte currentPage = page;
+						tasks.Add(GetPossiblyCompletedBadgeAppIDs(currentPage));
+					}
+
+					IList<HashSet<uint>?> results = await Utilities.InParallel(tasks).ConfigureAwait(false);
+
+					foreach (HashSet<uint>? result in results) {
+						if (result == null) {
+							return null;
+						}
+
+						firstPageResult.UnionWith(result);
+					}
+
+					return firstPageResult;
+			}
+		}
+
+		private async Task<HashSet<uint>?> GetPossiblyCompletedBadgeAppIDs(byte page) {
+			if (page == 0) {
+				throw new ArgumentNullException(nameof(page));
+			}
+
+			using IDocument? badgePage = await ArchiWebHandler.GetBadgePage(page).ConfigureAwait(false);
+
+			if (badgePage == null) {
+				ArchiLogger.LogGenericWarning(Strings.WarningCouldNotCheckBadges);
+
+				return null;
+			}
+
+			return GetPossiblyCompletedBadgeAppIDs(badgePage);
+		}
+
+		private HashSet<uint>? GetPossiblyCompletedBadgeAppIDs(IDocument badgePage) {
+			if (badgePage == null) {
+				throw new ArgumentNullException(nameof(badgePage));
+			}
+
+			List<IElement> linkElements = badgePage.SelectNodes("//a[@class='badge_craft_button']");
+
+			// We need to also select all badges that we have max level, as those will not display with a craft button
+			// Level 5 is maximum level for card badges according to https://steamcommunity.com/tradingcards/faq
+			linkElements.AddRange(badgePage.SelectNodes("//div[@class='badges_sheet']/div[contains(@class, 'badge_row') and .//div[@class='badge_info_description']/div[contains(text(), 'Level 5')]]/a[@class='badge_row_overlay']"));
+
+			if (linkElements.Count == 0) {
+				return new HashSet<uint>(0);
+			}
+
+			HashSet<uint> result = new HashSet<uint>(linkElements.Count);
+
+			foreach (string? badgeUri in linkElements.Select(htmlNode => htmlNode.GetAttribute("href"))) {
+				if (string.IsNullOrEmpty(badgeUri)) {
+					ArchiLogger.LogNullError(nameof(badgeUri));
+
+					return null;
+				}
+
+				// URIs to foil badges are the same as for normal badges except they end with "?border=1"
+				string? appIDText = badgeUri.Split('/', StringSplitOptions.RemoveEmptyEntries)[^1].Split('?', StringSplitOptions.RemoveEmptyEntries)[0];
+
+				if (!uint.TryParse(appIDText, out uint appID) || (appID == 0)) {
+					ArchiLogger.LogNullError(nameof(appID));
+
+					return null;
+				}
+
+				result.Add(appID);
+			}
+
+			return result;
+		}
+
+		private bool CompleteSetLootScheduled;
+		private const byte MinCardsPerBadge = 5;
+
+		private async Task SendCompletedSets() {
+			lock (SendCompleteSetsSemaphore) {
+				if (CompleteSetLootScheduled) {
+					return;
+				}
+
+				CompleteSetLootScheduled = true;
+			}
+
+			await SendCompleteSetsSemaphore.WaitAsync().ConfigureAwait(false);
+
+			try {
+				lock (SendCompleteSetsSemaphore) {
+					CompleteSetLootScheduled = false;
```

Comment:
Change `SendCompleteSetsSemaphore` and `CompleteSetLootScheduled` to more appropriate names based on the new config property.